lib/common/grpc-service.js

```diff
@@ -266,6 +268,25 @@ GrpcService.prototype.request = function(protoOpts, reqOpts, callback) {
     request: function(_, onResponse) {
       respError = null;
 
+      if (stream) {
+        service[protoOpts.method](reqOpts, grpcOpts)
+          .pipe(stream)
```

Comment:
It looks like we get `metadata` and `status`: https://github.com/grpc/grpc/blob/d79b7abf72f15362124dcb3b012bd06d6f331c23/src/node/src/client.js#L424-L425

Here's how I think this can look (omitting the other args to `retryRequest`):

```js
retryRequest({
  request: function() {
    return service[protoOpts.method](reqOpts, grpcOpts)
      .on('status', function(statusObj) {
        // not sure exactly what this callback receives,
        // but something like:
        this.emit('response', {
          code: statusObj.code
        });
      });
  }
});
```